org.osgi.test.junit5/src/main/java/org/osgi/test/junit5/bundle/BundleSource.java

```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright (c) OSGi Alliance (2020). All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.osgi.test.junit5.bundle;
+
+import java.lang.annotation.Documented;
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+import org.junit.jupiter.params.provider.ArgumentsSource;
+
+@Target({
+	ElementType.ANNOTATION_TYPE, ElementType.METHOD
+})
+@Retention(RetentionPolicy.RUNTIME)
+@Documented
+@ArgumentsSource(BundleArgumentsProvider.class)
+public @interface BundleSource {
+
+	/**
+	 * Optional SymbolicNameFilter used to filter Bundles by SymbolicName using
+	 * the an regex patterns.
+	 *
+	 * @return The symbolicNameFilter Strings.
+	 */
+	String[] symbolicNameFilter() default {};
+
+	/**
+	 * Optional stateFilter used to filter Bundles by state.
+	 *
+	 * @return The stateFilters.
+	 */
+	int[] stateFilter() default {};
+
+	/**
+	 * Filter string used to target a bundle by filtering the Bundle-Headers
+	 * Must use valid OSGi filter syntax.
+	 *
+	 * @return The filter string.
+	 */
+	String headerFilter() default "";
```

Comment:
So this one is actually an OSGi Filter expression but uses the same `Filter` name ending as the other elements which don't appear to be OSGi Filter expressions. The naming and documentation of these elements needs more clarity and detail.